exes/mqtt-broadcaster/inc/impl.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <sparkplug_b/sparkplug_b.pb.h>
+
+using org::eclipse::tahu::protobuf::Payload;
+using org::eclipse::tahu::protobuf::Payload_Metric;
+
+namespace tfc::mqtt::impl {
+
+struct signal_data {
+  tfc::ipc_ruler::signal information;
+  tfc::ipc::details::any_recv receiver;
+  std::optional<std::any> current_value;
+};
+
+// Create a timestamp in milliseconds from 1 Jan 1970 according to spec
+static auto timestamp_milliseconds() -> std::chrono::milliseconds {
+  using std::chrono::duration_cast;
+  using std::chrono::milliseconds;
+  using std::chrono::system_clock;
+  return duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+}
+
+/// This function converts tfc types to Spark Plug B types
+/// More information can be found (page 76) in the spec under section 6.4.16 data types:
+/// https://sparkplug.eclipse.org/specification/version/3.0/documents/sparkplug-specification-3.0.0.pdf
+auto type_enum_convert(tfc::ipc::details::type_e type) -> uint32_t {
+  using enum tfc::ipc::details::type_e;
+  switch (type) {
+    case unknown:
+      return 0;
+    case _bool:
+      return 11;
+    case _int64_t:
+      return 4;
+    case _uint64_t:
+      return 8;
+    case _double_t:
+      return 10;
+    case _string:
+    case _json:
+      return 12;
+  }
+  return 0;
+}
+
+auto set_value_payload(Payload_Metric* metric, std::any value, tfc::logger::logger& logger) -> void {
+  if (value.type() == typeid(bool)) {
+    metric->set_boolean_value(std::any_cast<bool>(value));
+  } else if (value.type() == typeid(std::string)) {
+    metric->set_string_value(std::any_cast<std::string>(value));
+  } else if (value.type() == typeid(uint64_t)) {
+    metric->set_long_value(std::any_cast<uint64_t>(value));
+  } else if (value.type() == typeid(int64_t)) {
+    metric->set_long_value(std::any_cast<int64_t>(value));
+  } else if (value.type() == typeid(double)) {
+    metric->set_double_value(std::any_cast<double>(value));
+  } else if (value.type() == typeid(float)) {
+    metric->set_float_value(std::any_cast<float>(value));
+  } else if (value.type() == typeid(uint32_t)) {
+    metric->set_int_value(std::any_cast<uint32_t>(value));
+  } else {
+    logger.error("Unknown type: {}", value.type().name());
```

Comment:
this is a bit confusing but I don't have an idea for fix at the moment other than custom error_code